src/CommunityToolkit/Xamarin.CommunityToolkit/Views/Popup/BasePopup.shared.cs

```diff
@@ -0,0 +1,175 @@
+﻿using System;
+using Xamarin.CommunityToolkit.Helpers;
+using Xamarin.Forms;
+
+namespace Xamarin.CommunityToolkit.UI.Views
+{
+	/// <summary>
+	/// The popup control's base implementation.
+	/// </summary>
+	[ContentProperty(nameof(Content))]
+	public abstract class BasePopup : VisualElement
```

Comment:
I fully support this idea, I don't think it is a solution that is fully supported by the current renderer architecture in Xamarin.Forms. I just tried implementing it and I ran into problems with the `ExportRenderer` statements. At that point in the code it is not clear what the type of `T` will be thus it creates a compiler error. this occurs because these statements are assembly level and are not scoped into any class that has a generic type argument.

This is invalid statement
```c#
[assembly: ExportRenderer(typeof(BasePopup<T>), typeof(PopupRenderer<T>))]
```

I did a little more prototyping and found that I could declare the renderer with the default type of `object` which allows it to compile. Since everything derives from `object` it does work, but I don't know if this really gets us type safety the way we want. **This seems like a hacky solution**, but it works.

My working solution
```c#
[assembly: ExportRenderer(typeof(BasePopup<object>), typeof(PopupRenderer<object>))]
```

This does in fact work and I was able to get a popup to render on the page. If we want to go down this path I think it would make sense to consolidate the `Popup` and `BasePopup` into one `Popup` class. They were split to handle issues with the native platform renderer not accepting generics. I think the current design of 2 classes is kind of confusing and would much rather have them put together.

Prior to moving forward with a change like this late in the code review process I would like feedback from all the other reviewers and/or someone from the Xamarin.Forms teams. Adding this change of defaulting `ExportRenderer` to `object` may have side-effects that we aren't thinking of at the moment.

cc: @AndreiMisiukevich @pictos @jsuarezruiz @jfversluis

Reply:
@ahoefling can you try this?
`[assembly: ExportRenderer(typeof(XPTO<>), typeof(XPTORenderer<>))]`

I just did a snippet here works.

Reply:
That's a good idea, I'll give that a try and report back

Reply:
I gave this a try and have some details to report. It does not work the way we were hoping 😢.

I merged the `BasePopup` and `Popup<T>` classes and then updated the `ExportRenderer`
```c#
[assembly: ExportRenderer(typeof(Popup<>), typeof(PopupRenderer<>)]
```

This technique causes an app crash when creating the custom renderer. It appears to be a problem that occurs in the Xamarin.Forms core code.

I don't think this is something we solve right now. I am more than willing to try something else if anyone has another idea.

Reply:
@brminnick @ahoefling I will admit I'm not fully aware of the context, but can we agree to revert to the (I assume) working first version, merge that and then investigate to see if we can improve the thing you're talking about. We probably need someone from the Forms team to have a look at what you're trying?

Since this won't be released, or well, whenever we want, we can still break stuff at will

Reply:
@jfversluis I prototyped the recommendation, but I never pushed it to the branch. I think the branch is good to merge as it is today.

Reply:
Gotcha! I feel we need to do a live stream with some kind of ribbon or big red button to merge this...

Anyway, @brminnick let us know your final thoughts about this thing and my comment and if you're OK.. DO IT 🙀

Reply:
Did I hear XCT web party?

Reply:
We can even live stream it on twitch 📡📺

Reply:
Sounds good to me! 

I had a hunch that `ExportRenderer` might be the blocker.

Thanks all!